### PR TITLE
Merge #15308 #15754 #15483 #16332

### PR DESCRIPTION
**Pull request by vijaydasmp**
cb3511b9d Add release notes for importing key origin info change (Andrew Chow)
4c75a69f3 Test importing descriptors with key origin information (Andrew Chow)
02d6586d7 Import KeyOriginData when importing descriptors (Andrew Chow)
3d235dff5 Implement a function to add KeyOriginInfo to a wallet (Andrew Chow)
eab63bc26 Store key origin info in key metadata (Andrew Chow)
345bff601 Remove hdmasterkeyid (Andrew Chow)
bac8c676a Add a method to CWallet to write just CKeyMetadata (Andrew Chow)
e7652d3f6 Add WriteHDKeypath function and move *HDKeypath to util/bip32.{h,cpp} (Andrew Chow)
c45415f73 Refactor keymetadata writing to a separate method (Andrew Chow)

Pull request description:

  This PR allows for key origin data as defined by the descriptors document to be imported to the wallet when importing a descriptor using `importmulti`. This allows the `walletprocesspsbt` to include the BIP 32 derivation paths for keys that it is watching that are from a different HD wallet.

  In order to make this easier to use, a new field `hdmasterkeyfingerprint` has been added to `getaddressinfo`. Additionally I have removed `hdmasterkeyid` as was planned. I think that this API change is fine since it was going to be removed in 0.18 anyways. `CKeyMetadata` has also been extended to store key origin info to facilitate this.

Tree-SHA512: 9c7794f3c793da57e23c5abbdc3d58779ee9dea3d53168bb86c0643a4ad5a11a446264961e2f772f35eea645048cb60954ed58050002caee4e43cd9f51215097

### CODE REVIEW

**Comment by Munkybooty** [comment]
Hey Vijay,

Is there an update on the status of this pr?

**Comment by vijaydasmp** [comment]
Hi Munkybooty,

I am not keeping well , couldn't devote time , probably work on it
tomorrow,  if you are blocked ,please feel free to take over

**Comment by vijaydasmp** [comment]
Hello @UdjinM6  please review

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for merging via merge commit

File: .travis.yml [hunk 1]
```diff
@@ -269,6 +269,6 @@ after_success:
         FILE_ENV="./ci/test/00_setup_env_amd64_nowallet.sh"
 
     - stage: test
-      name: 'macOS 10.12  [GOAL: deploy]'
+      name: 'macOS 10.10  [GOAL: deploy] [no functional tests]'
```
Comment (UdjinM6) [inline .travis.yml:272]: 15308: should keep 10.12

File: src/rpc/server.cpp [hunk 2]
```diff
@@ -234,7 +234,8 @@ static UniValue getrpcinfo(const JSONRPCRequest& request)
             "    \"method\"       (string)  The name of the RPC command \n"
             "    \"duration\"     (numeric)  The running time in microseconds\n"
             "   },...\n"
-            "  ]\n"
+            "  ],\n"
+            " \"logpath\": \"xxx\" (string) The complete file path to the debug log\n"
```
Comment (UdjinM6) [inline src/rpc/server.cpp:238]: 16332: requires 15483

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merge via merge commit

**Pull request merged**